perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1433 @@
+            The positions of the "new system"
+        use_dispersion_correction : bool, default False
+            Whether to use the long range correction in the custom sterics force. This is very expensive for NCMC.
+        """
+        self._topology_proposal = topology_proposal
+        self._old_system = copy.deepcopy(topology_proposal.old_system)
+        self._new_system = copy.deepcopy(topology_proposal.new_system)
+        self._old_to_hybrid_map = {}
+        self._new_to_hybrid_map = {}
+        self._hybrid_system_forces = dict()
+        self._old_positions = current_positions
+        self._new_positions = new_positions
+
+        self._use_dispersion_correction = use_dispersion_correction
+
+        self.softcore_alpha=0.5
+        self.softcore_beta=12*unit.angstrom**2
+
+        #prepare dicts of forces, which will be useful later
+        self._old_system_forces = {type(force).__name__ : force for force in self._old_system.getForces()}
+        self._new_system_forces = {type(force).__name__ : force for force in self._new_system.getForces()}
+
+        #get and store the nonbonded method from the system:
+        self._nonbonded_method = self._old_system_forces['NonbondedForce'].getNonbondedMethod()
+
+        #start by creating an empty system. This will become the hybrid system.
+        self._hybrid_system = openmm.System()
+
+        #begin by copying all particles in the old system to the hybrid system. Note that this does not copy the
+        #interactions. It does, however, copy the particle masses. In general, hybrid index and old index should be
+        #the same.
+        for particle_idx in range(self._topology_proposal.n_atoms_old):
+            particle_mass = self._old_system.getParticleMass(particle_idx)
+            hybrid_idx = self._hybrid_system.addParticle(particle_mass)
+            self._old_to_hybrid_map[particle_idx] = hybrid_idx
+
+            #If the particle index in question is mapped, make sure to add it to the new to hybrid map as well.
+            if particle_idx in self._topology_proposal.old_to_new_atom_map.keys():
+                particle_index_in_new_system = self._topology_proposal.old_to_new_atom_map[particle_idx]
+                self._new_to_hybrid_map[particle_index_in_new_system] = hybrid_idx
+
+        #Next, add the remaining unique atoms from the new system to the hybrid system and map accordingly.
+        #As before, this does not copy interactions, only particle indices and masses.
+        for particle_idx in self._topology_proposal.unique_new_atoms:
+            particle_mass = self._new_system.getParticleMass(particle_idx)
+            hybrid_idx = self._hybrid_system.addParticle(particle_mass)
+            self._new_to_hybrid_map[particle_idx] = hybrid_idx
+
+        #check that if there is a barostat in the original system, it is added to the hybrid.
+        #We copy the barostat from the old system.
+        if "MonteCarloBarostat" in self._old_system_forces.keys():
```

Comment:
What about other forces, like AndersonThermostat or something weird? Can you just check to see if there are no forces you don't explicitly handle here and throw an exception if either system contains something not explicitly copied? Otherwise we are going to run into trouble with this construction at some point, and it will take a long time to debug.